TheMovie/TheMovie/Scenes/Login/MainScreen/ViewModel/MainViewModel.swift

```diff
@@ -0,0 +1,28 @@
+//
+//  MainViewModel.swift
+//  TheMovie
+//
+//  Created by Jeiel Lima on 09/04/24.
+//
+
+import Foundation
+import GoogleSignIn
+
+class MainViewModel {
+    func loginWithGoogle(presentingViewController: UIViewController, completion: @escaping (User?, Error?) -> Void) {
+        GIDSignIn.sharedInstance.signIn(withPresenting: presentingViewController) { signInResult, error in
+            if let error = error {
+                completion(nil, error)
+                return
+            }
+            guard let signInResult = signInResult else {
+                completion(nil, nil)
+                return
+            }
+            let user = signInResult.user
+            let currentUser = User(username: user.profile?.name ?? "", email: user.profile?.email ?? "")
+            completion(currentUser, nil)
+        }
```

Comment:
Aqui faz parte da Main ou é exclusido para esse fim de conexão com a api? Se for, n seria um model.. e sim um Worker ou Repository seria mais interessante, e ai dentro da viewModel vc injetaria o worker e a ViewModel sim vc chamaria lá na ViewController.
Vamo trabalhar a questão de injeção de dependencias.

Outro ponto, se vc passa completion nil, ele vai crashar se n tiver info pra passar pra frente, como que trata essa parte? podemos criar um bug ou .. criar um improvement que seria alguma tela de erro pra gerenciar quando o usuário n tem essa info, tipo dar um alert e dar dismiss pra controller novamente uma vez q n conseguiu logar, mas pra completion o puro nil, vc tá dizendo q n tem nada pra passar pra frente, isso pode ser um problema.

Reply:
<a href="https://media2.giphy.com/media/v1.Y2lkPTc5MGI3NjExeWg3NndleDJlYWFsM2tsYjF1OXdzdDQxaGZsMHNsZnJlam8ydHlrayZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/wfS4vDyVsASQygl4mN/giphy.gif" target="_blank">Minha honesta reação</a></p>

Reply:
Kkkkkkkkk